packages/client/wallets/smart-wallet/src/blockchain/wallets/SendTransactionService.ts

```diff
@@ -0,0 +1,195 @@
+import { SmartAccountClient } from "permissionless";
+import { EntryPoint } from "permissionless/_types/types";
+import { SmartAccount } from "permissionless/accounts";
+import {
+    Abi,
+    Address,
+    BaseError,
+    Chain,
+    ContractFunctionRevertedError,
+    Hex,
+    PublicClient,
+    TransactionReceipt,
+    Transport,
+} from "viem";
+
+import { CrossmintSDKError, WalletErrorCode } from "@crossmint/client-sdk-base";
+
+export type TransactionServiceTransactionRequest = {
+    address: Address;
+    abi: Abi;
+    functionName: string;
+    args: any;
+    value?: bigint;
+};
+
+/**
+ * Error thrown when a transaction fails to send.
+ * @param viemError The error thrown by the viem client. See https://viem.sh/docs/glossary/errors.html
+ */
+export class EVMSendTransactionError extends CrossmintSDKError {
+    constructor(message: string, public readonly viemError: BaseError, code = WalletErrorCode.SEND_TRANSACTION_FAILED) {
+        super(message, code);
+    }
+}
+
+/**
+ * Error thrown when a transaction is sent successfully but fails to confirm.
+ * @param viemError The error thrown by the viem client. See https://viem.sh/docs/glossary/errors.html
+ */
+export class EVMSendTransactionConfirmationError extends EVMSendTransactionError {
+    constructor(
+        message: string,
+        public readonly viemError: BaseError,
+        code = WalletErrorCode.SEND_TRANSACTION_CONFIRMATION_FAILED
+    ) {
+        super(message, viemError, code);
+    }
+}
+
+/**
+ * Error thrown when a transaction simulation fails.
+ * @param viemError The error thrown by the viem client. See https://viem.sh/docs/glossary/errors.html
+ */
+export class EVMSendTransactionSimulationError extends EVMSendTransactionError {
+    constructor(
+        message: string,
+        public readonly viemError: BaseError,
+        code = WalletErrorCode.SEND_TRANSACTION_SIMULATION_FAILED
+    ) {
+        super(message, viemError, code);
+    }
+}
+
+/**
+ * Error thrown when a transaction fails due to a contract execution error.
+ * @param viemError The error thrown by the viem client. See https://viem.sh/docs/glossary/errors.html
+ * @param revertError The revert error from viem containing the reason for the revert.
+ * @param revertError.reason The reason for the revert.
+ * @param revertError.data The decoded revert error data.
+ * @example
+ * try {
+ *   await wallet.sendTransaction({
+ *     address: contractAddress,
+ *     abi,
+ *     functionName: "mintNFT",
+ *     args: [recipientAddress],
+ *   });
+ * } catch (e) {
+ *   if (e instanceof SendTransactionExecutionRevertedError) {
+ *     alert(`Transaction reverted: ${e.revertError.reason}`);
+ *   }
+ *   throw e;
+ * }
+ */
+export class EVMSendTransactionExecutionRevertedError extends EVMSendTransactionSimulationError {
+    constructor(
+        message: string,
+        public readonly viemError: BaseError,
+        public readonly revertError: ContractFunctionRevertedError,
```

Comment:
Rather than revertError here, I'd consider directly putting in relevant properties from it instead
  
```TS
data?: DecodeErrorResultReturnType | undefined
reason?: string | undefined
signature?: Hex | undefined
```

They
- are easier to access
- clear up any confusion from having two full viem errors, that could very well be the same.

Reply:
Also applies to `EVMSendTransactionSimulationError`